src/pages/LogsPage/TerminalRow.tsx

```diff
@@ -1,72 +1,97 @@
 import { Text } from '@status-im/components'
 import { Stack, XStack } from 'tamagui'
-import { faker } from '@faker-js/faker'
+import { LogType } from '../../constants'
 
-type DataType = {
-  option: string
-  description: string
+const colorMap = {
+  NTC: '#9b64b0',
+  DBG: '#7d828a',
+  INF: '#5ca8b4',
+  WRN: '#c9ad72',
+  ERR: '#bc5b5c',
+  FAT: '#b0686f',
 }
+
+const formatLog = (log: LogType, timestamps: boolean) => {
+  const { lvl, ts, msg, ...rest } = log
+
+  const logStyle = { color: colorMap[lvl] }
+  const tsStyle = { color: '#848994' }
+  const msgStyle = { color: '#a0a6b2', fontWeight: 'bold' }
+  const genericStyle = { color: '#5777c5' }
+
+  return (
+    <XStack space="$4">
+      <span style={{ ...logStyle }}>{lvl}</span>
+      {timestamps && <span style={{ ...tsStyle }}>{ts}</span>}
+      <span style={{ ...msgStyle }}>{msg}</span>
+      {Object.entries(rest).map(([key, value]) => {
+        if (value !== undefined && value !== null) {
+          return (
+            <div key={key}>
+              <span style={logStyle}>{key}=</span>
+              <span style={{ ...genericStyle }}>{`${
+                Array.isArray(value) ? value.join(', ') : value
```

Comment:
I suggest using `JSON.stringify` as it can efficiently handle logs of large objects and arrays without needing a special approach.
```suggestion
                                JSON.stringify(value)
```